src/sql/workbench/contrib/notebook/browser/cellToolbarActions.ts

```diff
@@ -140,6 +140,9 @@ export class CellToggleMoreActions {
 			if (i > 0 && actions[i] instanceof Separator && actions[i - 1] instanceof Separator) {
 				indexesToRemove.push(i);
 			}
+			if (i > 0 && actions[i] instanceof Separator && actions[i + 1] instanceof Separator) {
```

Comment:
This logic isn't correct. It only happens to work because there are multiple separators at the end so the 2nd to last one is "removed" by this check you're adding and then the last one is "removed" by the check above.

What we really care about though is removing the final item if it's a separator AFTER filtering out all the duplicates.

So just add that check and removal down below after the `if (indexesToRemove.length > 0) {` block.